app/src/main/java/com/guzelgimadieva/tasky/TaskyApp.kt

```diff
@@ -0,0 +1,57 @@
+package com.guzelgimadieva.tasky
+
+import android.content.Context
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import com.guzelgimadieva.tasky.agenda.ui.AgendaScreen
+import com.guzelgimadieva.tasky.authorization.ui.login.LoginScreen
+import com.guzelgimadieva.tasky.authorization.ui.register.RegisterScreen
+
+@Composable
+fun TaskyApp() {
+    val navController = rememberNavController()
+    val viewModel: AppViewModel = hiltViewModel()
+    val context = LocalContext.current
+
+    val refreshToken = context.getSharedPreferences(
+        context.getString(R.string.stored_refresh_token), Context.MODE_PRIVATE )
+        .all[context.getString(R.string.stored_refresh_token)].toString()
+    val userId = context.getSharedPreferences(
+        context.getString(R.string.stored_user_id), Context.MODE_PRIVATE)
+        .all[context.getString(R.string.stored_user_id)].toString()
+
+    if (refreshToken.isNotEmpty() && userId.isNotEmpty()) {
+        viewModel.getAccessToken(refreshToken, userId)
+    }
+
+    val startDestination = if (viewModel.isLoggedIn.collectAsState().value) {
```

Comment:
I'd recommend to use `collectAsStateWithLifecycle()` that since it's the safest way to collect flows in Android. For that, you'll need this dependency:
```
implementation "androidx.lifecycle:lifecycle-runtime-compose:2.x.x"
```
Then use it like this:
```
val state by viewModel.state.collectAsStateWithLifecycle()
```
If you have a simple flow it shouldn't make a difference, but as soon as you convert a normal `Flow` into a `StateFlow` with `stateIn`, the Flow operators will be executed even when the app is in the background (which you typically don't want when dealing with UI updates).